### ISSUE TITLE
Feature request: http(s) request support

### ISSUE THREAD

**Issue by kaiyuan01** [LingDong-/wenyan-lang#456]
To make wenyan-lang even more useful, we may consider consolidate a suite of APIs related to Chinese study to build a comprehensive SDK/API libraries (ku4, or packages), for example:

https://github.com/xenv/gushici

https://ctext.org/tools/api

The project may aim to provide APIs to access all those great resources already on the web. This cannot be done without a wrapper client class to access http(s) resources.

On top of that, we can build APIs such as:

1. Return a list of Chinese poems or sentences from Si Shu Wu Jing based on settings (weather, location, category etc)
2. Return the text given the book name and chapter;
3. Return the list of books/poems written, given the name of the author
4. Return the reconstructed pronunciation for a given character for a given period (Archaic, Middle or pre-Monern Chinese)

**Comment by LingDong-** [comment]
Hi @kaiyuan01 ,

I think HTTP requests will definitely make their way into the standard library; However specific API's to specific sites will probably be more suitable as 3rd party tools.

Thanks!

**Comment by lymslive** [comment]
While HTTP requests is surely lib, the ideal as @kaiyuan01 describ is a good application to show what wenyan-lang can do, that sound like a project related.

**Comment by cuixiping** [comment]
吾有一術名之曰「獲物」。欲行是術。必先得一言曰「源」。一術曰「理」是術曰。
    施「(x=>y=>fetch(x).then(e=>e.json()).then(y))」於「源」於「理」。
是謂「獲物」之術也。

吾有一術名之曰「獲言」。欲行是術。必先得一言曰「源」。一術曰「理」是術曰。
    施「(x=>y=>fetch(x).then(e=>e.text()).then(y))」於「源」於「理」。
是謂「獲言」之術也。

吾有一術名之曰「處理」。欲行是術。必先得一物。曰「甲」。是術曰。
    夫「甲」書之
是謂「處理」之術也。

施「獲言」於「「https://github.com/」」於「處理」。

**Comment by lymslive** [comment]
咦，这就实现了吗？那楼主可以开搞了哈。
我感觉这很有实践价值，用 wenyan-lang 帮助那些中国文化爱好者获取经典典籍知识，这很和谐。
似乎这比用 wenyan-lang 去重新实现《九章算术》之类的算法更具实用性，或更有利于 wenyan-lang 推广。

**Comment by cuixiping** [comment]
帮助那些中国文化爱好者获取经典典籍知识，跟 wenyan-lang 沒啥關係吧，直接讀書就可以了。
wenyan-lang是給文言愛好者兼編程愛好者用來編程的 😄

**Comment by lymslive** [comment]
我的意思是，wenyan-lang 可以自嗨，但是也要有梦想不满足于自嗨。
希望能通过类似的应用引起那些原来只喜欢中国传统文化但不会编程的文人的兴趣，进而学点wenyan-lang编程解决自己的其他问题。

**Comment by kaiyuan01** [comment]
Build extensive/useful packages to help non programmers to program using Wenyan-lang is a big deal. This helps promote Chinese culture especially for folks majoring in Chinese studies.

**Comment by LingDong-** [comment]
@kaiyuan01 , I agree these packages would be very useful, however they're too specific for the standard lib. Machine learning is useful, yet tensorflow is not part of python. Computer vision is useful, yet opencv is not part of C++. This is why we have 3rd party libraries.
The HTTP request library on the other hand, is suitable for the stdlib.
Thanks!
